etl/steps/data/garden/homicide/2023-01-04/unodc.py

```diff
@@ -76,13 +76,55 @@ def harmonize_countries(df: pd.DataFrame) -> pd.DataFrame:
 
 
 def clean_data(df: pd.DataFrame) -> pd.DataFrame:
+    # Making the units more obvious
+    df["unit_of_measurement"] = df["unit_of_measurement"].replace(
+        ["Counts", "Rate per 100,000 population"], ["Homicide count", "Homicides per 100,000 population"]
+    )
+    # Splitting the data into that which has the totals and that which is disaggregated by mechanism
+    df_mech = df[df["dimension"] == "by mechanisms"]
 
-    df = df[["country", "year", "unit_of_measurement", "value"]]
+    df_mech = create_mechanism_df(df_mech)
 
-    df = (
-        df.pivot(index=["country", "year"], columns="unit_of_measurement", values="value")
-        .reset_index()
-        .rename_axis(None, axis=1)
-    )
+    df_tot = df[df["dimension"] == "Total"]
+
+    df_tot = create_total_df(df_tot)
+
+    df = pd.merge(df_mech, df_tot, how="outer", on=["country", "year"])
 
     return df
+
+
+def create_total_df(df_tot: pd.DataFrame) -> pd.DataFrame:
+    """Create the total homicides dataframe where we will have total homicides/homicide rate
+    disaggregated by age and sex
+    """
+    df_tot = df_tot.drop(columns=["region", "subregion", "indicator", "dimension", "category", "source"])
+    # There are some duplicates when sex is unknown so let's remove those rows
+    df_tot = df_tot[df_tot["sex"] != "Unknown"]
+    # Make it more obvious what total age and total sex means
+    df_tot["age"] = df_tot["age"].replace("Total", "All ages")
+    df_tot["sex"] = df_tot["sex"].replace("Total", "Both sexes")
```

Comment:
You can try to use `pd.Series.map` with argument `na_action="ignore"` instead, I believe it runs faster. Something like:

```python
df_tot["age"] = df_tot["age"].map({"Total", "All ages"}, na_action="ignore")
df_tot["sex"] = df_tot["sex"].map({"Total", "Both sexes"}, na_action="ignore")
```

Reply:
I think this replaces anything not in the dict with NaN, which I don't want. I just want to replace some of the values in that column with a more descriptive value.

Reply:
Ah if I have ` df_tot["age"].map({"Total", "All ages"}, na_action="ignore").fillna(df_tot['age']` then it fills any non-replaced values with the original column